src/cgutils.cpp

```diff
@@ -2704,7 +2704,7 @@ static jl_cgval_t emit_new_struct(jl_codectx_t &ctx, jl_value_t *ty, size_t narg
             if (lt->isVectorTy() || jl_is_vecelement_type(ty)) { // maybe also check the size ?
                 init_as_value = true;
             }
-            else if (tracked.count) {
+            else if (tracked.count == 0) {
```

Comment:
```suggestion
            else if (tracked.count != 0) {
```

We want to keep this as a FCA if there are tracked objects. We generally don't want to `init_as_value` (though `mark_julia_type` would sink it to the stack) if we don't have to.

Reply:
I'm confused about what happens below this:
```
            else if (init_as_value) {
                 ...
            }
            else {
                strct = emit_static_alloca(ctx, lt);
                if (tracked.count)
                    undef_derived_strct(ctx.builder, strct, sty, tbaa_stack);
            }
```
If `tracked.count` was nonzero, we would have set `init_as_value`, so it looks like the `undef_derived_strct` is unreachable.